### DISCUSSION TITLE
Select the newly created file when copy/pasting a file

### DISCUSSION THREAD

**Discussion by josesilveiraa07** [zed-industries/zed#24704] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

When a file is copied and pasted, the old file remains selected, but I think we all are used to have the newly created file selected so we can rename it or do whatever we want with it.

**Comment by notpeter** [comment]
Yep. I can reproduce this issue.

**Comment by romainreignier** [comment]
I have encountered the same issue with ZED preview 0.154.0.
The new file is selected in the project panel but not opened, which VSCode does by default.
Should I open a separate issue for this?

**Comment by SomeoneToIgnore** [comment]
If you have the repro steps, that would be great to do.

**Comment by romainreignier** [comment]
Sure:
1. Select a file in the project panel
2. Right click
3. Click on "Duplicate"
4. The new file (with " copy" suffix) is created, selected in the left panel but not opened in the editor.

The same with:
1. Select a file in the project panel
2. Right click
3. Click on "Copy"
4. Right click in the project panel
5. Click "Paste"re

ZED preview 0.154
OS : Ubuntu 22.04 KDE

https://github.com/user-attachments/assets/2f362a1b-c57e-4603-8c78-f1d79ca65b4c

**Comment by burrbull** [comment]
Same issue. After copy+paste new file is not in focus. (Zed: v0.167.1, ubuntu 24.04)

**Comment by notpeter** [comment]
This has been fixed. When you paste or duplicate a file the new filename is focused with rename active so you can type a new filename.

Thanks for reporting.

**Comment by romainreignier** [comment]
Actually, the issue was that the new file is not opened as a new tab in the editor (so we can start modifying it). If we edit the current file opened, it is the original file, not the copy.
It seems to me that this issue is not fixed, at least not in version 0.172.11.
